### PR TITLE
aws-sqs-queue: Support metadata via TriggerAuthentication

### PR DESCRIPTION
**Pull request by obfu5c8**
Updates `aws-sqs-queue` scaler to use `GetFromAuthOrMeta` instead of `TriggerMetadata` to allow for passing metadata to the scaler via TriggerAuthentication.

### Checklist

- [ ] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [ ] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #

Relates to #

### CODE REVIEW

**Comment by JorTurFer** [comment]
Hi
Why do you want to treat as sensitive data things like `activationQueueLength` (which isn't sensitive at all)?

**Comment by obfu5c8** [comment]
> Hi Why do you want to treat as sensitive data things like `activationQueueLength` (which isn't sensitive at all)? is it for reusing the configuration? I ask because we are thinking about introducing any reuse mechanism for triggers

I don't, but since TriggerAuthentication provides an abstracted standard for defining params for any/all trigger types it seems like the sensible way forward rather than each trigger implementing some/all/none in its own way.

**Comment by JorTurFer** [comment]
> The sensitivity wasn't a driver, but (in my very narrow case) having a standardised way to configure via env vars

What do you think about this @kedacore/keda-core-contributors ?

**Comment by tomkerkhove** [comment]
If sensitivity is not the driver, then I don't think it makes sense to be honest. What is your use-case exactly? Probably `FromEnv` parameters are a better fit

**Review by zroubalik — COMMENTED** [review]
Agree with @tomkerkhove on this one.

**Comment by obfu5c8** [comment]
The use case is standardisation across scalers I think. If I'm rolling out keda as a tool across my org, it's much easier to say "and this is how you can dynamically configure any of the scalers" rather than "this scaler has this special fromEnv param, whereas this one does it slightly differently, and because this value is sensitive you need to configure it over here instead".

As a concrete use case, lets say I'm deploying the same service in multiple environments/namespaces. I could put a same-name-but-differently-configured TriggerAuthentication resource in each ns to change the `awsRoleArn` depending on the environment, and the ScaledObject resource wouldn't need changing (since it just references the TriggerAuthentication res). But also it's a different `awsRegion`, so separately we need to change the ScaledObject resource as well.

The TriggerAuthentication resource handles this abstraction nicely as it offers multiple options for sourcing the values decoupled from the scaler implementation itself. The fact that it's not directly 'authentication' related seems to be a semantic argument about the resource name rather than any particular misuse of functionality (unless I'm missing something deeper TA resources do under the hood).

In short, using a TA (or something like it) gives users the choice and options to decide where to source their config from, and if a new source was added in the future it wouldn't require updating every Scaler type for it to become available.

**Comment by JorTurFer** [comment]
> In short, using a TA (or something like it) gives users the choice and options to decide where to source their config from, and if a new source was added in the future it wouldn't require updating every Scaler type for it to become available.

The problem is that conceptually, TriggerAuthentication implies sensitive data.  What do you think if we would introduce another CRD similar to the trigger authentication but for any (not sensitive) data? I'm thinking in a trigger template with support to env variables. Could this match your requirements? (and this is totally hypothetical, other maintainers should agree with it)

**Comment by obfu5c8** [comment]
I guess  this conversation should probably be moved into an issue since the PR itself isn't destined to be merged.

> What do you think if we would introduce another CRD similar to the trigger authentication but for any (not sensitive) data?

If we're talking hypothetical alternatives, how about the following?
- Rename `TriggerAuthentication` to `TriggerConfiguration` to remove the conceptual implications of sensitivity.
- Allow trigger definitions to define a stack/list of `TriggerConfiguration` resources (instead of/as well as the current single `TriggerAuthentication`) that would be applied as overlays to each other (with direct trigger params layered on top at the end)

This would support the current use-case ("A different team owns/maintains the auth credentials resources") as well as giving flexibility in configuration methods going forward. If every scaler bought into the new model it'd mean consistency across implementations, and no more `fromEnv` parameters added ad-hoc

**Comment by tomkerkhove** [comment]
Let's close this PR and move the conversation to a new issue to discuss this as we need to think this through.

> Rename TriggerAuthentication to TriggerConfiguration to remove the conceptual implications of sensitivity.

This is what I do *not* want to do. We need to emphasize what are secrets and what are not; otherwise people will misuse them.

**Comment by zroubalik** [comment]
Let's go a different direction.
